### PR TITLE
support ESM builds

### PR DESCRIPTION
**Pull request by luu-alex**
## Description
#6127 
This PR aims to allow esm based projects to build web3 without errors, so we don't need to run esm builds with the flag `--experimental-specifier-resolution=node`

The issue here was that relative imports had no file extension so it couldn't resolve properly. Having the file extension end with .js doesn't change how typescript compiles but there needed to be changes on [jest ](https://github.com/web3/web3.js/pull/6131/files#diff-1dfad82ce3093edb53297fd2c11bf0e5e3340f1795453bcb2d7bdbc35e17ec93)and our [linter](https://github.com/web3/web3.js/pull/6131/files#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R14). 
Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Comment by Sammiegirl411** [comment]
Yes

File: packages/web3-core/src/index.ts [hunk 1]
```diff
@@ -15,17 +15,17 @@ You should have received a copy of the GNU Lesser General Public License
...
-export * from './web3_subscription_manager';
-export * from './web3_subscriptions';
-export * from './web3_context';
-export * from './web3_batch_request';
-export * from './utils';
-export * from './types';
-export * from './formatters';
-export * from './web3_promi_event';
-export * from './web3_event_emitter';
+export * from './web3_config.js';
```
Comment (jdevcs) [inline packages/web3-core/src/index.ts:18]: Thanks @luu-alex , could you run blackbox tests on this branch ( locally or in github ) for looking blackbox is also passing.

**Review by avkos — APPROVED** [review]

**Comment by jdevcs** [comment]
LGTM,

**Review by jdevcs — APPROVED** [review]

File: packages/web3-core/src/index.ts (hunk 1 above)
Comment (luu-alex) [reply packages/web3-core/src/index.ts:18]: 👍  https://github.com/web3/web3.js/actions/runs/5134874349/jobs/9239775016 all other ci tests were run in a previous commit and it passed

**Review by Spacesai1or — APPROVED** [review]

**Pull request merged**